src/documentation/markdown.rs

```diff
@@ -169,11 +168,29 @@ impl<'a> PlainOverlay<'a> {
                     }
                 }
                 Event::Text(s) => {
+                    match current_link_type {
```

Comment:
A `if let Some(x) = y {` combined with a `match y { .. }` might simplify this, by removing to match against all `Some(_)` wrapped variants.